core/src/main/scala/cats/std/map.scala

```diff
@@ -64,4 +66,72 @@ trait MapInstances extends algebra.std.MapInstances {
 
       override def isEmpty[A](fa: Map[K, A]): Boolean = fa.isEmpty
     }
+
+  implicit def mapMonoid[K, V: Semigroup]: MapMonoid[K, V] =
+    new MapMonoid[K, V]
+
+  implicit def mapGroup[K, V: Group]: MapGroup[K, V] =
+    new MapGroup[K, V]
+}
+
+class MapMonoid[K, V](implicit V: Semigroup[V]) extends Monoid[Map[K, V]]  {
+  def empty: Map[K, V] = Map.empty
+
+  def combine(x: Map[K, V], y: Map[K, V]): Map[K, V] =
+    MapMethods.addMap(x, y)(V.combine)
+}
+
+class MapGroup[K, V](implicit V: Group[V]) extends MapMonoid[K, V] with Group[Map[K, V]] {
+  def inverse(x: Map[K, V]): Map[K, V] =
+    x.map { case (k, v) => (k, V.inverse(v)) }
+
+  override def remove(x: Map[K, V], y: Map[K, V]): Map[K, V] =
+    MapMethods.subtractMap(x, y)(V.remove)(V.inverse)
+}
+
+object MapMethods {
+  def initMutableMap[K, V](m: Map[K, V]): mutable.Map[K, V] = {
+    val result = mutable.Map.empty[K, V]
+    m.foreach { case (k, v) => result(k) = v }
+    result
+  }
+
+  def wrapMutableMap[K, V](m: mutable.Map[K, V]): Map[K, V] =
```

Comment:
Are `initMutableMap` and `wrapMutableMap` methods that Cats wants to expose in its kernel API? As they return mutable values, they stick out when compared to the rest of the Cats API. Since binary compatibility is a big concern in `kernel`, I'm wondering if it would be better to make this `private` for now, as it's always easier to open them up later than it is to remove them once they are exposed in a release.